server/controllers/profile.js

```diff
@@ -0,0 +1,124 @@
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+const generateToken = require("../utils/generateToken");
+
+// @route POST /profiles
+// @desc Given parameters passed in create a profile
+// @access Private
+
+ exports.createProfile = asyncHandler(async (req, res, next) => {
+
+    const { firstName, lastName, description, email, phoneNumber, availability:{startDate,endDate,daysOfWeek} } = req.query;
```

Comment:
Why are we using query parameters, shouldn't we be getting these fields from the request body (req.body) ?
Ditto throughout!